execution_engine/src/core/engine_state/execution_result.rs

```diff
@@ -315,6 +315,9 @@ impl ExecutionResult {
     ///
     /// The effects that are produced as part of this process would subract `max_payment_cost` from
     /// account's main purse, and add `max_payment_cost` to proposer account's balance.
+    ///
+    /// NOTE: This function assumes that `max_payment_cost <= account_main_purse_balance`.
+    /// Not obeying this restriction is a programming error with undefined behavior.
```

Comment:
While it is currently possible to underflow as indicated by the comment, wouldn't it be worthwhile to implement `Motes::checked_sub()` and use that instead of the bare subtraction in `make_payment_error_effects`? @Fraser999

Reply:
@dwerner @Fraser999 addressed in ac7706d77c601fd75d8f09030f64369aec2a5a23 and ticketed bigger scope of removal of unchecked ops in https://github.com/casper-network/casper-node/issues/2103.